pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1016,7 +1016,7 @@ func buildUpstreamClusterTLSContext(opts *buildClusterOpts, tls *networking.Clie
 		}
 
 	case networking.ClientTLSSettings_SIMPLE:
-		if tls.CredentialName != "" {
+		if tls.CredentialName != "" && proxy.Type == model.Router  {
```

Comment:
so in this case - now the sidecar gets no config for tls. Is that expected?

Reply:
It gets some TLS config, but not the SDS config over the CredentialName when the originator of the DestinationRule is a sidecar proxy, which is the desired outcome

Reply:
I am a bit concerned a user will set up a DR and be surprised when there is absolutely no validation going on. Or for mTLS, confused we aren't sending a cert either

Reply:
This is documented in the DR API though that credentialName is only when proxy is of type Gateway. If the user instead of setting up CredentialName, sets up Cert/Key/CA in TLS for DR, validation will happen

Reply:
We can log an error if you think its required when tls.CredentialName != "" && proxy.Type == model.Sidecar

Reply:
same concern here

Reply:
On second thought, it's not a good idea to log error when tls.CredentialName != "" && proxy.Type == model.Sidecar because DestinationRule is exported to all namespaces by default and if sidecar makes a call through Gateway, the sidecar will log error when it's actually a non issue